src/BinSkim.Driver/AnalysisSummaryExtractor.cs

```diff
@@ -51,17 +57,44 @@ public static void UpdateBuildPipelineInfo(AnalysisSummary summary)
             // https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml
             if (summary != null)
             {
-                try
-                {
-                    summary.BuildDefinitionId = Environment.GetEnvironmentVariable(BuildDefinitionIdVar);
-                    summary.BuildDefinitionName = Environment.GetEnvironmentVariable(BuildDefinitionNameVar);
-                    summary.BuildRunId = Environment.GetEnvironmentVariable(BuildDefinitionRunIdVar);
-                }
-                catch (SecurityException)
+                summary.BuildDefinitionId = ExtractValueFromEnvironmentVariable(BuildDefinitionIdVar);
+                summary.BuildDefinitionName = ExtractValueFromEnvironmentVariable(BuildDefinitionNameVar);
+                summary.BuildRunId = ExtractValueFromEnvironmentVariable(BuildDefinitionRunIdVar);
+
+                summary.RepositoryId = ExtractValueFromEnvironmentVariable(RepositoryIdVar);
+                summary.RepositoryName = ExtractValueFromEnvironmentVariable(RepositoryNameVar);
+
+                summary.ProjectId = ExtractValueFromEnvironmentVariable(ProjectIdVar);
+                summary.ProjectName = ExtractValueFromEnvironmentVariable(ProjectNameVar);
+
+                summary.OrganizationId = ExtractValueFromEnvironmentVariable(OrganizationIdVar);
+                summary.OrganizationName = ExtractValueFromEnvironmentVariable(OrganizationNameVar);
+                summary.OrganizationName = summary.OrganizationName.Replace("https://dev.azure.com/", string.Empty).TrimEnd('/');
```

Comment:
>TrimEnd('/') [](http://example.com/codeflow?start=116&length=12)

Question, will the summary.OrganizationName only possible with
[https://dev.azure.com/A/](https://dev.azure.com/A/)
or also possible with
[https://dev.azure.com/A/B/](https://dev.azure.com/A/B/)

Reply:
will only contain A/